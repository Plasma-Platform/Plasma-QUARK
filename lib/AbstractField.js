'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _TextField = require('./TextField.js');

var _TextField2 = _interopRequireDefault(_TextField);

var _TextArea = require('./TextArea.js');

var _TextArea2 = _interopRequireDefault(_TextArea);

var _utils = require('./utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InputElement = function (_Component) {
  (0, _inherits3.default)(InputElement, _Component);

  function InputElement(props, context) {
    (0, _classCallCheck3.default)(this, InputElement);

    var _this = (0, _possibleConstructorReturn3.default)(this, (InputElement.__proto__ || (0, _getPrototypeOf2.default)(InputElement)).call(this, props, context));

    _this.state = {
      value: _this.props.value || '',
      filled: _this.props.filled || !!_this.props.value || false,
      focused: !!_this.props.focused,
      isValid: null,
      animated: false,
      notificationText: _this.props.notificationText || '',
      limitCounter: _this.props.maxLength
    };

    _this.componentDidMount = function () {
      _this.inputElement = _this.comp.input.target.input;
      _this.animationTarget = _this.props.componentType === 'textarea' ? _this.comp.input.target.textarea : _this.inputElement;
      _this.inputDOMElement = _reactDom2.default.findDOMNode(_this.animationTarget);
      if (_this.props.autofocus) {
        _this.focus();
      }

      _this.inputDOMElement.addEventListener('animationend', function (event) {
        event.stopPropagation();
        /* @todo: найти более элегантное решение
         для отслеживания полного завершения анимации из нескольких кейфреймов */
        if (event.animationName === 'hideErrorField' || event.animationName === 'hideSuccessField') {
          _this.setState({
            animated: false
          });
        }
      }, false);

      if (_this.props.limitCounter) {
        _this.refreshInputCounter();
      }
    };

    _this.componentWillReceiveProps = function (nextProps) {
      if (nextProps.value !== _this.props.value) {
        _this.setState({
          value: nextProps.value,
          filled: true
        });
      }
    };

    _this.componentDidUpdate = function () {
      if (_this.state.isValid === false) {
        _this.comp.input.showNotification();
      } else {
        _this.comp.input.hideNotification();
      }
    };

    _this.focus = function (event) {
      _this.setState({
        focused: true
      });
      _this.inputElement.focus(event);
    };

    _this.setValidationStatus = function (status, notificationText) {
      _this.handleValidation({
        status: status,
        message: notificationText
      });
      _this.activateAnimation();
    };

    _this.getValidationStatus = function () {
      return _this.state.isValid;
    };

    _this.getValue = function () {
      return _this.state.value;
    };

    _this.changeFieldType = function () {
      var newType = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'text';

      _this.inputDOMElement.setAttribute('type', newType);
    };

    _this.onChange = function (event) {
      if (typeof _this.props.onChange === 'function') {
        _this.props.onChange(event);
      }
      _this.setState({
        value: event.target.value,
        filled: !!event.target.value,
        isValid: null
      });

      _this.refreshInputCounter();
    };

    _this.onFocus = function (event) {
      if (!_this.state.focused) {
        if (typeof _this.props.onFocus === 'function') {
          _this.props.onFocus(event);
        }

        _this.oldValue = _this.state.value;
        _this.setState({
          focused: true,
          filled: !!event.target.value
        });
      }
    };

    _this.onBlur = function (event) {
      if (typeof _this.props.onBlur === 'function') {
        _this.props.onBlur(event);
      }

      var trimmed = _this.state.value.trim();

      if (!_this.inputElement.value) {
        _this.setState({
          filled: false,
          focused: false
        });
      } else {
        _this.setState({
          focused: false,
          value: trimmed
        });
      }
      if (typeof _this.props.onValidate === 'function' && (_this.state.isValid === null || _this.oldValue !== _this.inputElement.value)) {
        _this.props.onValidate();
        _this.activateAnimation();
      }
    };

    _this.handleValidation = function (data) {
      _this.setState({
        isValid: data.status,
        notificationText: data.message
      });
    };

    _this.resetValidationStatus = function () {
      _this.setState({
        isValid: undefined
      });
    };

    _this.activateAnimation = function () {
      _this.setState({
        animated: true
      });
    };

    _this.refreshInputCounter = function () {
      var textareaElement = _this.comp.input.target.input;

      if (textareaElement.value === 0) {
        return;
      }

      var currentValue = _this.props.maxLength - (textareaElement ? textareaElement.value.length : 0);
      var currentMaxValue = Math.max(0, currentValue);

      _this.setState({
        limitCounter: currentMaxValue
      });
    };

    _this.oldValue = '';
    _this.inputElement = null;
    _this.inputDOMElement = null;
    _this.textComponent = props.componentType === 'textarea' ? _TextArea2.default : _TextField2.default;
    _this.component = (0, _utils.connectNotificationTextField)(_this.textComponent);
    return _this;
  }

  // приводим к нулю, возможные отрицательные значения, которые появляются в счетчике
  // при монтировании компонента с заполненным значением, превышающим установвленный лимит


  (0, _createClass3.default)(InputElement, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var DecoratedTextField = this.component;

      return _react2.default.createElement(DecoratedTextField, (0, _extends3.default)({}, this.props, {
        ref: function ref(comp) {
          return _this2.comp = comp;
        },
        value: this.state.value,
        filled: this.state.filled,
        focused: this.state.focused,
        limitCounter: this.state.limitCounter,
        isValid: this.state.isValid,
        animated: this.state.animated,
        onFocus: this.onFocus,
        onBlur: this.onBlur,
        onChange: this.onChange,
        changeFieldType: this.changeFieldType,
        resetValidationStatus: this.resetValidationStatus,
        notification: { code: this.props.notificationType || 'N2B',
          text: this.state.notificationText,
          maxWidth: this.props.notificationMaxWidth
        }
      }));
    }
  }]);
  return InputElement;
}(_react.Component);

InputElement.propTypes = {
  id: _react.PropTypes.string,
  componentType: _react.PropTypes.oneOf(['textfield', 'textarea']),
  sizeType: _react.PropTypes.string.isRequired,
  placeholder: _react.PropTypes.string,
  value: _react.PropTypes.string,
  disabled: _react.PropTypes.bool,
  filled: _react.PropTypes.bool,
  focused: _react.PropTypes.bool,
  notificationText: _react.PropTypes.string,
  maxLength: _react.PropTypes.number
};
InputElement.defaultProps = {
  notificationAlt: { status: false }
};
exports.default = InputElement;