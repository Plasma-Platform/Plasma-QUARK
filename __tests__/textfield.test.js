/* global expect:false, test:false, it:false, describe:false, jest:false, beforeEach:false */

import React from 'react';
import {shallow, mount} from 'enzyme';
import TextField from '../src/components/textFields';
import TextFieldPresent from '../src/components/TextField.jsx';
import AbstractField from '../src/components/AbstractField.jsx';

describe('1 TextField high level components [F1, F2, F3, F4] testing', () => {
  let PROPS = {
    type: ['email', 'password', 'text', 'search']
  };

  let callback;
  let tfCount;
  let textFields = [];

  beforeEach(() => {
    callback = jest.fn();
    tfCount = Object.keys(TextField).length;
    for (let i = 1; i <= tfCount; i++) {
      let componentType = `F${i}`;
      let component = TextField[componentType];
      textFields.push(component);
    }
  });

  it('1.1 makes snapshots for All 4 sizes of TF with default parameters', () => {
    textFields.forEach((Component) => {
      let node = shallow(<Component />).html();
      expect(node).toMatchSnapshot();
    });
  });

  it('1.2 makes snapshots for All 4 sizes of TF * each 4 input types', () => {
    textFields.forEach((Component) => {
      PROPS.type.forEach((type) => {
        let node = shallow(<Component type={type} />).html();
        expect(node).toMatchSnapshot();
      });
    });
  });

  it('1.3 checks that textField.value returns a value of input', () => {
    textFields.forEach((Component) => {
      let node = mount(<Component value='test value' />);
      let value = node.component.getInstance().value;
      expect(value).toEqual('test value');
    });
  });

  it('1.4 checks that componentType of TextField is "textfield"', () => {
    textFields.forEach((Component) => {
      let node = mount(<Component />);
      let type = node.find('TextField').first().props().componentType;
      expect(type).toEqual('textfield');
    });
  });

  it('1.5 checks that sizeType prop points to right size', () => {
    for (let i = 1; i <= tfCount; i++) {
      let size = `F${i}`;
      let Component = TextField[size];
      let node = mount(<Component />);
      let sizeType = node.find('TextField').first().props().sizeType;
      expect(sizeType).toEqual(size);
    }
  });

  it('1.8 performs focus in field using focus() method', () => {
    textFields.forEach((Component) => {
      let node = mount(<Component />);
      node.node.focus();
      expect(node.find('TextField').props().focused).toEqual(true);
    });
  });

  describe('1.2 Events', () => {
    it('1.2.1 calls onFocus callback', () => {
      textFields.forEach((Component) => {
        let node = mount(<Component onFocus={callback} />);
        node.find('input').simulate('focus');
        expect(callback).toBeCalled();
      });
    });

    it('1.2.2 calls onBlur callback', () => {
      textFields.forEach((Component) => {
        let node = mount(<Component onBlur={callback} />);
        node.find('input').simulate('blur');
        expect(callback).toBeCalled();
      });
    });

    it('1.2.3 calls onChange callback', () => {
      textFields.forEach((Component) => {
        let node = mount(<Component onChange={callback} />);
        node.find('input').simulate('change');
        expect(callback).toBeCalled();
      });
    });
  });
});

describe('2 AbstractField testing', () => {
  let testNode;
  let input;
  let callback;

  beforeEach(() => {
    callback = jest.fn();
    testNode = mount(<AbstractField onFocus={callback} onBlur={callback} onChange={callback}/>);
    input = testNode.find('input').first();
  });

  describe('2.1 Focus testing', () => {
    it('2.1.1 defines onFocus callback presence', () => {
      expect(testNode.props().onFocus).toBeDefined();
    });

    it('2.1.2 calls onFocus callback', () => {
      input.simulate('focus');
      expect(callback).toBeCalled();
    });

    it('2.1.3 checks that onFocus added "text-field_focused" class into wrapping element', () => {
      input.simulate('focus');
      let classes = testNode.find('.abstract-field').first().props().className;
      expect(classes).toMatch('_focused');
    });

    it('2.1.4 checks that onFocus made state.focused = true', () => {
      input.simulate('focus');
      expect(testNode.state('focused')).toEqual(true);
    });
  });

  describe('2.2 Blur testing', () => {
    it('2.2.1 defines onBlur callback presence', () => {
      expect(testNode.props().onBlur).toBeDefined();
    });

    it('2.2.2 calls onBlur callback', () => {
      input.simulate('blur');
      expect(callback).toBeCalled();
    });

    it('2.2.3 checks that onBlur removed "text-field_focused" class from wrapping element', () => {
      input.simulate('focus');
      input.simulate('blur');
      let classes = testNode.find('.abstract-field').first().props().className;
      expect(classes).not.toMatch('_focused');
    });

    it('2.2.4 checks that onBlur made state.focused = false', () => {
      input.simulate('focus');
      input.simulate('blur');
      expect(testNode.state('focused')).toEqual(false);
    });
  });

  describe('2.3 Change testing', () => {
    it('2.3.1 defines onChange callback presence', () => {
      expect(testNode.props().onChange).toBeDefined();
    });

    it('2.3.2 calls onChange callback', () => {
      let newVal = '777';
      input.simulate('change', {target: {value: newVal}});
      expect(input.props().value).toMatch(newVal);
    });

    it('2.3.3 changes field\'s value with non-empty string, cause state.filled=true', () => {
      let newVal = '7';
      input.simulate('change', {target: {value: newVal}});
      expect(testNode.state('filled')).toEqual(true);
    });

    it('2.3.4 cleans field\'s value, cause state.filled=false', () => {
      input.simulate('change', {target: {value: ''}});
      expect(testNode.state('filled')).toEqual(false);
    });
  });

  describe('2.4 Functions ', () => {
    it('2.4.1 receives the value of input', () => {
      testNode.setProps({'value': 'test'});
      let value = testNode.node.getValue();
      expect(value).toEqual('test');
    });

    it('2.4.2 sets up validation status to invalid and starts animation', () => {
      testNode.node.setValidationStatus(false, 'Error');
      expect(testNode.state('isValid')).toEqual(false);
      expect(testNode.state('animated')).toEqual(true);
    });

    it('2.4.3 changes field type', () => {
      let newType = 'email';
      testNode.node.changeFieldType(newType);
      expect(input.node.type).toEqual(newType);
    });

    it('2.4.4 animates field', () => {
      testNode.node.activateAnimation();
      expect(testNode.state('animated')).toEqual(true);
      let wrapperClasses = testNode.find('.abstract-field').props().className;
      expect(wrapperClasses).toMatch('animated');
    });

    it('2.4.5 resets validation status', () => {
      testNode.node.resetValidationStatus();
      expect(testNode.state('isValid')).toEqual(undefined);
      let wrapperClasses = testNode.find('.abstract-field').props().className;
      expect(wrapperClasses).not.toMatch(/valid |invalid /);
    });

    it('2.4.6 trims string on blur', () => {
      let testNode = mount(<AbstractField />);
      let input = testNode.find('input');
      let newVal = '   spaces';
      input.simulate('focus');
      input.node.value = newVal;
      input.simulate('change', {target: {value: newVal}});
      input.simulate('blur');
      expect(input.node.value).toEqual(newVal.trim());
    });
  });
});

describe('3 TextField Presentational testing', () => {
  let callback;
  beforeEach(() => {
    callback = jest.fn();
    jest.useFakeTimers();
  });

  it('3.1 checks that type of input is "text" by default', () => {
    let node = mount(<TextFieldPresent />);
    let type = node.props().type;
    expect(type).toEqual('text');
  });

  it('3.2 checks that input type=password has an password reveal icon', () => {
    let node = shallow(<TextFieldPresent type="password" />);
    let reveal = node.find('EyePasswordIndicator').first();
    expect(reveal).toBeDefined();
  });

  it('3.3 shows tooltip ', () => {
    let node = mount(<TextFieldPresent />);
    node.node.showTooltip(callback);
    expect(callback).not.toBeCalled();
    jest.runAllTimers();
    expect(callback).toBeCalled();
    expect(callback.mock.calls.length).toBe(1);
  });

  it('3.4 hides password and tooltip', () => {
    let node = mount(<TextFieldPresent changeFieldType={callback} />);
    let hideTooltip = jest.fn();
    node.node.hidePasswordAndTooltip(hideTooltip);
    expect(hideTooltip).not.toBeCalled();
    jest.runAllTimers();
    expect(hideTooltip).toBeCalled();
    expect(callback).toBeCalledWith('password');
  });

  it('3.4.1 hides password and tooltip with force cleaning timer', () => {
    let node = mount(<TextFieldPresent changeFieldType={callback} />);
    node.instance().showTimer = true;
    node.node.hidePasswordAndTooltip();
    expect(callback).toBeCalled();
  });

  it('3.5 simulate mouseenter on reveal icon', () => {
    let node = mount(<TextFieldPresent type="password" />);
    let reveal = node.find('EyePasswordIndicator').first();
    node.node.showTooltip = jest.fn(node.node.showTooltip);
    reveal.simulate('mouseover');
    expect(node.node.showTooltip).toBeCalled();
  });

  it('3.6 simulate touchstart on reveal icon', () => {
    let node = mount(<TextFieldPresent type="password" />);
    let reveal = node.find('EyePasswordIndicator').first();
    node.node.showTooltip = jest.fn(node.node.showTooltip);
    reveal.simulate('touchstart');
    expect(node.node.showTooltip).toBeCalled();
  });

  it('3.7 simulate mouseleave on reveal icon', () => {
    let node = mount(<TextFieldPresent type="password" />);
    let reveal = node.find('EyePasswordIndicator').first();
    node.node.hidePasswordAndTooltip = jest.fn(node.node.hidePasswordAndTooltip);
    reveal.simulate('mouseleave');
    expect(node.node.hidePasswordAndTooltip).toBeCalled();
  });

  it('3.8 simulate touchend on reveal icon', () => {
    let node = mount(<TextFieldPresent type="password" />);
    let reveal = node.find('EyePasswordIndicator').first();
    node.node.hidePasswordAndTooltip = jest.fn(node.node.hidePasswordAndTooltip);
    reveal.simulate('touchend');
    expect(node.node.hidePasswordAndTooltip).toBeCalled();
  });
});
