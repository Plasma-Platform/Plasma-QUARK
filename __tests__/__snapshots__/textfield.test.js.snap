exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.1 makes snapshots for All 4 sizes of TF with default parameters 1`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.1 makes snapshots for All 4 sizes of TF with default parameters 2`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.1 makes snapshots for All 4 sizes of TF with default parameters 3`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.1 makes snapshots for All 4 sizes of TF with default parameters 4`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 1`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"email\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 2`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"password\" value=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 3`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 4`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"search\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 5`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-letter\"></span><input type=\"email\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 6`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-key\"></span><input type=\"password\" value=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 7`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 8`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-magnifier\"></span><input type=\"search\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 9`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"email\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 10`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"password\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 11`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 12`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"search\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 13`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-letter\"></span><input type=\"email\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 14`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-key\"></span><input type=\"password\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 15`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 16`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-magnifier\"></span><input type=\"search\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 17`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"email\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 18`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"password\" value=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 19`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 20`] = `"<div class=\"abstract-field text-field text-field_f1\"><input type=\"search\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 21`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-letter\"></span><input type=\"email\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 22`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-key\"></span><input type=\"password\" value=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 23`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 24`] = `"<div class=\"abstract-field text-field text-field_f2\"><span class=\"text-field__type-icon icon icon-magnifier\"></span><input type=\"search\" value=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 25`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"email\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 26`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"password\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 27`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 28`] = `"<div class=\"abstract-field text-field text-field_f3\"><input type=\"search\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 29`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-letter\"></span><input type=\"email\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 30`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-key\"></span><input type=\"password\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><div><span class=\"text-field__notification-icon icon\"></span><span class=\"text-field__notification-icon icon password-toggle icon-eye\"></span></div><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 31`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon\"></span><input type=\"text\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;

exports[`1 TextField high level components [F1, F2, F3, F4] testing 1.2 makes snapshots for All 4 sizes of TF * each 4 input types 32`] = `"<div class=\"abstract-field text-field text-field_f4\"><span class=\"text-field__type-icon icon icon-magnifier\"></span><input type=\"search\" value=\"\" placeholder=\"\" class=\"text-field__input field-style\"/><span class=\"text-field__notification-icon icon\"></span><label class=\"text-field__label\"></label></div>"`;
